core/src/Dscp/Crypto/Impl.hs

```diff
@@ -75,6 +80,27 @@ type PublicKey = AbstractPK SigScheme
 type SecretKey = AbstractSK SigScheme
 type Signature a = AbstractSig SigScheme a
 
+data DscpSigScheme
+data instance SD.PublicKey DscpSigScheme = DscpPK PublicKey
```

Comment:
Also, I do not understand this design decision. Why is it an open data family? Why not a closed one? Why not a type family?

Reply:
> Why not a type family?

There were some problems with `VerifySign` type class and making `PublicKey` and `Signature` data families was the easiest (while still seamless) solution.

> Why not a closed one?

Well, because there could be a universe of various signature schemes.

Reply:
No, i mean the `class SigScheme where data SSPub, data SSpriv`... Currently i can define public key without defining a secret one, which is weird.

Reply:
> Currently i can define public key without defining a secret one, which is weird.

Because for the sake of validation (which is main concern of snowdrop to date) we don't need a secret key.

Reply:
Yes, but the signature scheme with pk without sk doesn't make any sense at all. It doesn't have anything to do with what you ask in your methods, it's just there should be a single place to define pk and sk. 

Let's end this conversation, we got sidetracked a bit.